src/utils.js

```diff
@@ -10,30 +10,53 @@ export function getNetworkName(chainId) {
 
 export function rpcResult(response) {
   // Some providers don’t wrap the response
-  if (response && response.jsonrpc) {
+  if (response && 'jsonrpc' in response) {
     if (response.error) {
       throw new Error(response.error)
     }
-    return response.result
+    return response.result || ''
   }
-  return response
+  return response || ''
+}
+
+async function sendCompat(ethereum, method, params) {
+  // Old MetaMask versions are using sendAsync(), and web3.js 1.x is
+  // overwriting send() with sendAsync(), despite them being incompatible.
+  // There is no way to detect that send() follows the correct implementation,
+  // so we use sendAsync() when it is available.
+  if (ethereum.sendAsync && ethereum.selectedAddress) {
```

Comment:
Do we need `ethereum.selectedAddress`?

Reply:
Yes, `sendAsync` requires `from` to be set: https://github.com/aragon/use-wallet/blob/web3-js-compat/src/utils.js#L30

Since we are using a soon-deprecated API, I thought we might as well use this and avoid a request or to pass the account.